src/sentry/utils/os_normalization.py

```diff
@@ -0,0 +1,25 @@
+from __future__ import absolute_import
+import re
+
+# Environment.OSVersion (GetVersionEx) or RuntimeInformation.OSDescription, on Windows
+windows = re.compile('^(Microsoft )?Windows (NT )?(?P<version>\d+\.\d+\.\d+).*$')
+# Environment.OSVersion or RuntimeInformation.OSDescription (uname)
+# on Mono and CoreCLR on macOS, iOS, Linux, etc
+uname = re.compile('^(?P<name>[a-zA-Z]+) (?P<version>\d+\.\d+\.\d+(\.[1-9]+)?).*$')
+
+
+# If there's no name and version, attempts to infer from raw_description
+def normalize(data):
+    raw_description = data.get('raw_description', None)
```

Comment:
_nit:_ `data.get('foo')` will default to `None`, so you could omit the second parameter.